src/x11/request/input.rs

```diff
@@ -1439,7 +1440,15 @@ impl<const KEYSYMS_PER_KEYCODE: usize> Writable for ChangeKeyboardMapping<KEYSYM
 	#[allow(clippy::cast_possible_truncation)]
 	fn write_to(&self, buf: &mut impl BufMut) -> WriteResult {
 		// Limit `buf` by the length (converted to bytes).
+		#[cfg(not(feature = "big-requests"))]
 		let buf = &mut buf.limit(usize::from(self.length()) * 4);
+		#[cfg(feature = "big-requests")]
+		let buf = &mut buf.limit({
+			let Ok(size) = (self.length() * 4).try_into() else {
+				return Err(WriteError::Other(Box::new("Unable to allocate buffer")));
+			};
+			size
+		});
```

Comment:
```suggestion
		let size = match usize::try_from(self.length()) {
			Ok(size) => size * 4,

			Err(error) => return Err(WriteError::Other(Box::new(error))),
		};
		let buf = &mut buf.limit(size);
```
...I mean the only case where there could be an error is on a 16-bit platform, and I'm not particularly sure anyone is gonna be running this on a 16-bit platform lol...

Doing the multiplication after converting to `usize` because, well, it's almost definitely the case that `usize` is either 32-bit (meaning no difference) or 64-bit (meaning it's possible that you couldn't multiply the length by 4 when it's `u32` but you can when it's `usize`).  Only case where that wouldn't be true is 16-bit (can you even run an X server on a 16-bit machine?? I suppose it is _technically_ possible to run a barebones X client on a 16-bit machine that connects to an X server remotely anyway... not that there's any reason to support that, only that we might as well if it isn't a burden of any kind).